File: cpp/392.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.empty()) return true;
        if(t.empty()) return false;

        int i = 0;
        for(auto& c : t) {
            if(c == s[i])
                i++;
            if(i == s.size())
                return true;
        }
        return i == s.size();
    }
};

File: python/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0
        if(len(s) == 0):
            return True
        if(len(t) == 0):
            return False
        for c in t:
            if(c == s[i]):
                i+=1
            if(i==len(s)):
                return True

        return i==len(s)

